### PR TITLE
Do not offer External channel in tabs where it is not relevant

### PR DESCRIPTION
**Pull request by CyrilleB79**
### Link to issue number:
Closes #15039
### Summary of the issue:
External appears in channel filter list even in tabs where it is not relevant, i.e. in Available add-ons and Updatable add-ons.
### Description of user facing changes
Remove these External channel from the channel filter list in these tabs.
### Description of development approach
See code.
### Testing strategy:
Manual testing.
### Known issues with pull request:
None

### Change log
None for such a small fix.

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

### CODE REVIEW

**Comment by lukaszgo1** [comment]
> External appears in channel filter list even in tabs where it is not relevant, i.e. in Available add-ons and Updatable add-ons.

Why you consider 'external' to be irrelevant in the 'updateable' tab? As far as I can tell add-ons in that tab can be either installed from the store, or installed from the add-on bundle, but with a more recent version present in the store.###

**Comment by CyrilleB79** [comment]
> Why you consider 'external' to be irrelevant in the 'updateable' tab? As far as I can tell add-ons in that tab can be either installed from the store, or installed from the add-on bundle, but with a more recent version present in the store.

*Because the channel visible in the Updatable add-ons tab is not the channel of the installed add-on but the channel through which a new version can be offered from the store.

E.g. say you have installed Speech logger 23.3.08 externally. Checking the updatable add-ons:
* if you select stable channel, you are offered the stable version from the store, i.e. 23.3.09
* if you select dev channel, you are offered the dev version from the store, i.e. 23.3.109  sélectionné
* if you select external channel, you are not offered anything since the store does not offer anything through external channel by definition

To summarize, the column "Channel" in this tab should be understood as "Available channel", not as "Installed channel".

**Review by seanbudd — APPROVED** [review]
Thanks @CyrilleB79
